lib/github_service.rb

```diff
@@ -12,8 +12,17 @@ module GithubService
   #
   class << self
     def add_comments(fq_repo_name, issue_number, comments)
-      Array(comments).each do |comment|
+      # add_status requires first comment due to its URL
+      Array(comments).map do |comment|
         add_comment(fq_repo_name, issue_number, comment)
+      end.first
+    end
+
+    # https://octokit.github.io/octokit.rb/Octokit/Client/Statuses.html#create_status-instance_method
+    def add_status(comstat, comment)
+      unless comstat.nil?
+        comstat["options"]["target_url"] = comment["html_url"]
+        create_status(comstat["repo"], comstat["sha"], comstat["state"], comstat["options"])
```

Comment:
Again, you're sending arguments from a single hash, these things can be simplified.